### PR TITLE
Add block setters for consensus type

### PR DESCRIPTION
**Pull request by terencechain**
Pat of #11629 

This PR adds block setters which is very useful for block construction for validators. We no longer need to duplicate every protobuf-specific functions for every hard fork. 

Note: Setting a field in a block as currently designed is not thread-safe, the only caller is the proposer at the block construction stage. This is similar to what we have today with setting state root. I'll be happy to change this to a thread-safe model with lock if there's feedback

### CODE REVIEW

File: consensus-types/blocks/getters.go [hunk 1]
```diff
@@ -751,6 +786,11 @@ func (b *BeaconBlockBody) Graffiti() [field_params.RootLength]byte {
 	return b.graffiti
 }
 
+// SetGraffiti sets the graffiti in the block.
+func (b *BeaconBlockBody) SetGraffiti(g []byte) {
```
Comment (nisdas) [inline consensus-types/blocks/getters.go:790]: this isnt a getter

**Review by nisdas — COMMENTED** [review]
I think the title of the PR should indicate that this is also implements block setters

**Comment by terencechain** [comment]
> I think the title of the PR should indicate that this also implements block setters

Big typo for the title. It's meant to write block setters lol

File: beacon-chain/core/transition/transition_no_verify_sig_test.go [hunk 2]
```diff
@@ -39,7 +39,9 @@ func TestExecuteStateTransitionNoVerify_FullProcess(t *testing.T) {
 	require.NoError(t, err)
 	require.NoError(t, beaconState.SetSlot(beaconState.Slot()-1))
 
-	nextSlotState, err := transition.ProcessSlots(context.Background(), beaconState.Copy(), beaconState.Slot()+1)
```
Comment (potuz) [inline beacon-chain/core/transition/transition_no_verify_sig_test.go:42]: Why these changes?

File: beacon-chain/core/transition/transition_no_verify_sig_test.go [hunk 3]
```diff
@@ -95,7 +97,9 @@ func TestExecuteStateTransitionNoVerifySignature_CouldNotVerifyStateRoot(t *test
...
```
Comment (potuz) [inline beacon-chain/core/transition/transition_no_verify_sig_test.go:98]: Why these changes?

File: consensus-types/blocks/execution.go [hunk 4]
```diff
@@ -155,6 +155,16 @@ func (e executionPayload) WithdrawalsRoot() ([]byte, error) {
 	return nil, ErrUnsupportedGetter
 }
 
+// PbV1 --
+func (e executionPayload) PbV1() (*enginev1.ExecutionPayload, error) {
```
Comment (potuz) [inline consensus-types/blocks/execution.go:159]: All these PbVx from the different targets are not tested.

File: consensus-types/blocks/getters.go [hunk 5]
```diff
@@ -731,6 +731,41 @@ func (b *BeaconBlock) AsSignRequestObject() (validatorpb.SignRequestObject, erro
 	}
 }
 
+func (b *BeaconBlock) Copy() (interfaces.BeaconBlock, error) {
```
Comment (potuz) [inline consensus-types/blocks/getters.go:734]: This is not tested

File: beacon-chain/core/transition/transition_no_verify_sig_test.go (hunk 3 above)
Comment (terencechain) [reply beacon-chain/core/transition/transition_no_verify_sig_test.go:98]: Sorry. It's meant for another PR

File: consensus-types/blocks/execution.go (hunk 4 above)
Comment (rkapka) [inline consensus-types/blocks/execution.go:159]: Do we really want to name things `PbV1`/`PbV2` instead of `Pb`/`PbCapella` or `PbBellatrix`/`PbCapella`? It might be misleading because all protobufs are under the `enginev1` package.

File: consensus-types/blocks/setters.go [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+// This function is not thread safe, it is only used during block creation.
+func (b *SignedBeaconBlock) SetBlock(blk interfaces.BeaconBlock) error {
+	copied, err := blk.Copy()
+	if err != nil {
+		return err
+	}
+	b.block.slot = copied.Slot()
+	b.block.parentRoot = copied.ParentRoot()
+	b.block.stateRoot = copied.StateRoot()
+	b.block.stateRoot = copied.StateRoot()
```
Comment (rkapka) [inline consensus-types/blocks/setters.go:26]:
```suggestion
	b.block.stateRoot = copied.StateRoot()
```
